feature_engine/categorical_encoders.py

```diff
@@ -827,3 +829,144 @@ def transform(self, X):
             X[self.variables] = X[self.variables].astype('O')
 
         return X
+
+
+class DecisionTreeCategoricalEncoder(BaseEstimator, TransformerMixin):
```

Comment:
Shouldn't we be using the base categorical transformer? If that is the case it is missing a fit method that could be added (the other categorical encoders could also use this). The base also requires an encoder dict, while we're currently using the pipeline as the encoder. We can add an additional `encoder_dict`  attribute to the class as well.

Reply:
Good point! I thought of the baseNumerical, because it has all the code we need in the transform method. But it will be confusing for future developers (or even myself) if we inherit for a categorical encoder a numerical base transformer. So I would say keep as it is.
The base categorical has a lot of functionality that we don't need here. And I am not sure we can reconstruct the original variable after first encoding and then using trees. So I would not include the inverse_transform for this transformer. And thus, we do not need a new encoder_dict.

But I think it would be important to help users navigate the result of this transformer from the docstrings.

Thanks a lot for this! Well spotted!